api/src/main/java/com/epam/pipeline/manager/git/GitlabClient.java

```diff
@@ -224,44 +199,17 @@ public GitCredentials buildCloneCredentials(boolean useEnvVars, Long duration) {
                                                           boolean recursive) throws GitClientException {
         try {
             String projectId = makeProjectId(namespace, projectName);
-
-            Map<String, Object> params = new HashMap<>();
-            if (StringUtils.isNotBlank(path)) {
-                params.put("path", path);
-            }
-            if (StringUtils.isNotBlank(revision)) {
-                params.put("ref_name", revision);
-            }
-            params.put("recursive", recursive);
-
-            String url = addUrlParameters(String.format(GIT_GET_SOURCES_ROOT_URL, gitHost, projectId), params);
-            URI uri = new URI(url);
-
-            LOGGER.trace("Getting repository contents on path {}, URL: {}", path, uri);
-
-            RestTemplate template = new RestTemplate();
-            ResponseEntity<List<GitRepositoryEntry>> sourcesResponse = template.exchange(uri, HttpMethod.GET,
-                    getAuthHeaders(), new ParameterizedTypeReference<List<GitRepositoryEntry>>() {});
-
-            if (sourcesResponse.getStatusCode() == HttpStatus.OK) {
-                return sourcesResponse.getBody();
-            } else {
-                throw new UnexpectedResponseStatusException(sourcesResponse.getStatusCode());
-            }
-        } catch (UnsupportedEncodingException | URISyntaxException e) {
+            return gitLabApi.getRepositoryTree(projectId, path, revision, recursive).execute().body();
+        } catch (IOException e) {
             throw new GitClientException(e);
         }
     }
 
-    public GitProject createTemplateRepository(Template template,
-                                               String name,
-                                               String description,
-                                               boolean indexingEnabled,
-                                               String hookUrl)
-            throws GitClientException {
+    public GitProject createTemplateRepository(Template template, String name, String description,
+                                               boolean indexingEnabled, String hookUrl) throws GitClientException {
         try {
             return createGitProject(template, description, convertPipeNameToProject(name), indexingEnabled, hookUrl);
```

Comment:
Maybe we can unify error handling and throw `GitlabClientException` from private methods and not handle them in public methods? Currently some methods fully rely on `execute` error handling, others catch some exceptions and rethrow them as `GitlabClientException`.